homeassistant/components/alexa/handlers.py

```diff
@@ -377,6 +455,8 @@ async def async_api_set_percentage(hass, config, directive, context):
         service = fan.SERVICE_SET_PERCENTAGE
         percentage = int(directive.payload["percentage"])
         data[fan.ATTR_PERCENTAGE] = percentage
+    else:
```

Comment:
We can invert the check above and raise if true. Then we can outdent below.

Reply:
I originally left it out, thinking that the PR was maybe big enough.
Since you requested it, I implemented for the four corresponding methods in e3fb5ede51058708641ce0ad2adf33dc9ca403b5